src/situations/commun/infra/depot_ressources.js

```diff
@@ -1,14 +1,19 @@
-function chargeurAudio (src) {
-  const audio = new window.Audio(src);
+const chargeurAudio = function (src, timeout = 2000) {
+  const audio = new window.Audio();
 
-  return new Promise((resolve, reject) => {
+  const promesse = new Promise((resolve, reject) => {
+    const cloneur = () => new window.Audio(src);
     audio.addEventListener('canplaythrough', () => {
-      resolve(() => new window.Audio(src));
+      resolve(cloneur);
     });
 
+    setTimeout(() => { resolve(cloneur); }, timeout);
```

Comment:
je comprends bien le comportement de timeout, mais que se passe-t-il si la promesse a déjà été résolue ? Elle se résout 2 fois ?

Reply:
Bonne question. Je en sais pas mais en tout cas le `Promise.all(this.promesses)` est validé des le premier "resolve" et donc on attends pas forcément 2 secondes.